Lesson9/main.js

```diff
@@ -0,0 +1,47 @@
+const ul = document.getElementById("js-ul");
+const loading = document.getElementById("js-loading")
+const features = [{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
+{to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}];
+
+const addLoading = () =>{
+  const loadingImg = document.createElement('img');
+  loadingImg.src = 'loading.gif';
+  loadingImg.id = "loading-img";
+  loading.appendChild(loadingImg);
+};
+
+const removeLoading = () =>{
+  const removeImg = document.getElementById('loading-img')
+  removeImg.remove();
+};
+
+const createLists = (features) =>{
+  const fragment = new DocumentFragment();
+
+  for (let feature of features ){
+      const li = document.createElement('li');
+      const a = document.createElement('a');
+      const img = document.createElement('img');
+      a.href = feature.to;
+      a.textContent = feature.text;
+      img.src = feature.img;
+      img.alt = feature.alt;
+
+      fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin', img);
+    };
+
+    ul.appendChild(fragment);
+};
+
+async function asyncFeatures(){
+  addLoading();
+  let loadingTimeout = new Promise((resolve) =>{
+    setTimeout(() => resolve(features), 3000)
+  });
```

Comment:
IMO
Why not cut this part out as a single function?

Q
And, why do you use let?

Reply:
As per your feedback, I put those lines into a one function. 
Also there is no specific intention of using let, I just followed one of the articles from references as example. 
Changed let to const as the variable 's value does not need to be re-assigned. 
Please check my latest commit 28cc47a6943db8e2233c1d0dbabc798be71deb5f